tsdb/chunkenc/chunk.go

```diff
@@ -72,10 +73,23 @@ type Appender interface {
 }
 
 // Iterator is a simple iterator that can only get the next value.
+// Iterator iterates over the samples of a time series.
 type Iterator interface {
+	// Next advances the iterator by one.
+	Next() bool
+	// Seek advances the iterator forward to the first sample with the timestamp equal or greater than t.
+	// If current sample found by previous `Next` or `Seek` operation already has this property, Seek has no effect.
+	// Seek returns true, if such sample exists, false otherwise.
+	// Iterator is exhausted when the Seek returns false.
+	// TODO(bwplotka): Verify above statements on all implementations with unit test.
+	Seek(t int64) bool
+	// At returns the current timestamp/value pair.
+	// At returns (math.MinInt64, 0.0) before the iterator has advanced.
+	// TODO(bwplotka): Verify above statement on all implementations with unit test.
 	At() (int64, float64)
+	// Err returns the current error.
+	// Err can return undefined value before the iterator has advanced.
```

Comment:
I would be specific and say "before calling Next or Seek", because "iterator has advanced" is likely not true. But that's really just nitpicking.

Reply:
It is fair, fixed it (: